### DISCUSSION TITLE
The build system asks for C17. Why does the code look barely complient with the 1989 standard?

### DISCUSSION THREAD

**Discussion by roflcopter4** [winsiderss/systeminformer#1887] [Questions / Answers] [answered]
I mean really. No mixed declarations and code? Usage of stupid macros like `VOID`? Dang near every single include is specified with angle brackets, which I can't even begin to understand. The reserved namespace is violated routinely, and for absolutely no reason. You guys know that structure tags are in their own namespace and can be the same as a typedef right? What's up with all this? Can I fix it? I removed every single violating namespace, enum, and union tag in a fork but I'm hesitant to make such a large pull request.

**Comment by jxy-s** [comment, marked as answer]
Our headers follow a standard and style quite similar to Microsoft's. Our source also adheres to these comparable standards and style. While we appreciate feedback, at this time, we do not require additional suggestions. We kindly ask for any further detailed or pedantic remarks to be directed elsewhere. Additionally, please reconsider the manner in which you present questions, comments, or suggestions in a public forum.

**Reply by roflcopter4** [reply]
Microsoft uses this style because they are implementing system files. They may use the reserved namespace because of this. They unnecessarily give structure tags different names to typedefs out of a traditional support for ancient, pre-standard C compilers. You are not writing a system library, and you don't support old C compilers. As I mentioned, you request the 2017 C standard.

**Reply by jxy-s** [reply]
> You are not writing a system library

It seems obvious to me that you have neglected to read the manual: https://github.com/winsiderss/systeminformer/blob/master/phnt/README.md

> you don't support old C compilers. As I mentioned, you request the 2017 C standard.

The compilers and standards work fine for our projects. It sounds like the issue may be specific to your setup or machine. If you can provide details about what's broken for you, the experts here may be willing to help. Simply yelling "compliance" into the void will not yield any results.
